control-plane/stor-port/src/types/v0/transport/policies.rs

```diff
@@ -0,0 +1,86 @@
+use crate::types::v0::transport::Nexus;
+
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+
+/*
+    This file defines the policies that control plane uses to make
+    decisions about rebuild workflow behaviour in case of a child
+    becoming faulted. We can logically provide multiple diferent policies.
+    These policies are represented as an enum, since only one of the policy
+    can be attached to a given volume/nexus at a time, and this attachment
+    will be done during provisioning. Once attached, the further behaviour
+    of the policy can vary depending on runtime state of the nexus.
+    Any new policy introduced in future will have to implement Policy trait
+    for defining the policy behaviour.
+*/
+
+// time constants, in seconds
+const TWAIT_SPERF: std::time::Duration = Duration::new(600, 0);
+const TWAIT_SAVAIL: std::time::Duration = Duration::new(300, 0);
+const TWAIT_ZERO: std::time::Duration = Duration::new(0, 0);
+// 100GiB = 100 * 1024 * 1024 * 1024 bytes
+const VOL_SIZE_100GIB: u64 = 107374182400;
+
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, Eq, PartialEq)]
+// SystemPerf policy
+pub struct SystemPerf {}
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, Eq, PartialEq)]
+// SystemAvail policy
+pub struct SystemAvail {}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone, Copy)]
+pub enum RuleSet {
+    SystemPerf(SystemPerf),
+    SystemAvail(SystemAvail),
+}
+
+impl RuleSet {
```

Comment:
The logic should probably live in the core agent itself?

Reply:
Are you suggesting that policies.rs be moved under agents/core hierarchy somewhere? Perhaps we can, I'm not too clear on control-plane code organisation yet.

Reply:
Yeah, that sounds good. Since we have started to have policy driven approaches and that will only increase in future, better to have seperate modules.

Reply:
So while rearranging this, realised that it'll be a problem to move everything into core agent because when we add a new RuleSet type field to Nexus, it'll not be identified in store-port.
So either:
1. Need to split the definitions and implementation. Keeping definitions in stor-port/transport, and implementation in core-agent.
2. OR, not binding the policy to nexus statically, rather call assign() every time a child goes faulted and we need to get the timed-wait value. It's not a costly function call so should be ok. 

Exploring option 2.

Reply:
Updated the PR with 2.) as mentioned above.
The assign() call exists in the faulted_child_wait() path but we don't use it for returning value now. We'll simply return default 10mins. Once we decide to handle based on size and other factors, we can just remove the default path.